driver-core/src/main/com/mongodb/client/model/expressions/Expressions.java

```diff
@@ -184,6 +185,35 @@ public static <T extends Expression> ArrayExpression<T> ofArray(final T... array
         });
     }
 
+    public static <T extends Expression> EntryExpression<T> ofEntry(final String k, final T v) {
+        Assertions.notNull("k", k);
+        Assertions.notNull("v", v);
+        return new MqlExpression<>((cr) -> {
+            BsonDocument document = new BsonDocument();
+            document.put("k", new BsonString(k));
+            document.put("v", extractBsonValue(cr, v));
+            return new AstPlaceholder(new BsonDocument("$literal",
+                    document.toBsonDocument(BsonDocument.class, cr)));
```

Comment:
I also removed the `$literal` wrapper: `return new AstPlaceholder(document);`.

This allows entries to be created from expressions, which seems fine since we allow this for ofArray.